src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Orchestration.cs

```diff
@@ -56,11 +55,15 @@ public partial class ReindexDurableFunction
             List<int> queryTagKeys = queryTags.Select(x => x.Key).ToList();
             if (queryTags.Count > 0)
             {
-                List<WatermarkRange> batches = await GetBatchesAsync(context, input.Completed, logger);
+                IReadOnlyList<WatermarkRange> batches = await context.CallActivityWithRetryAsync<IReadOnlyList<WatermarkRange>>(
```

Comment:
this logic is much more cleaner than the other one https://github.com/microsoft/dicom-server/pull/976, maybe we should check this one in at first.

Reply:
Looking at my other PR, I think that makes sense too